pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -449,7 +439,33 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
         return outcome_map;
     }
 
+    void
+    setDeviceSeed(const std::optional<std::size_t> &deviceSeed = std::nullopt) {
+        this->DeviceSeed = deviceSeed;
+    }
+
+    void setDeviceSeedFromPRNG(std::mt19937 *gen) {
+        if (gen != nullptr) {
+            this->setDeviceSeed((*gen)());
+        }
+    }
+
+    void activateDeviceSeed() {
+        std::size_t seed = this->DeviceSeed.has_value()
+                               ? this->DeviceSeed.value()
+                               : this->generateRandomSeed();
+        rng.seed(seed);
+    }
+
   private:
+    /**
+     * @brief Generate a random seed
+     *
+     */
+    std::size_t generateRandomSeed() {
```

Comment:
Lightning follows the tailing style for return data-types:
```suggestion
    auto generateRandomSeed() -> std::size_t {
```

Reply:
Removed